astroid/interpreter/_import/spec.py

```diff
@@ -423,6 +425,18 @@ def _find_spec_with_path(
     raise ImportError(f"No module named {'.'.join(module_parts)}")
 
 
+def cache_modpath(func):
+    def wrapper(*args):
+        key = ".".join(args[0])
+        if key not in modpath_cache:
+            modpath_cache[key] = func(*args)
```

Comment:
maybe this cache also needs to be cleared there ?
 https://github.com/pylint-dev/astroid/blob/4ccb9c42734a1c06cccb7bd3e0e381a70eabadfd/astroid/manager.py#L437

Reply:
Thanks for pointing that out. The cache is now cleared from there and I also renamed variables a little bit and made the code look more like existing cache code (e.g. the inference tip cache).